src/parser.rs

```diff
@@ -1,35 +1,63 @@
 use html5ever::{self, Attribute};
 use html5ever::tree_builder::{TreeSink, NodeOrText, QuirksMode};
 use std::borrow::Cow;
+use std::fs::File;
+use std::path::Path;
+use std::io::Read;
 use string_cache::QualName;
 use typed_arena::Arena;
 
+
 use tree::Node;
+use std::option;
+
+pub struct Html<F = IgnoreParseErrors> where F: FnMut(Cow<'static, str>) {
+    opts: ParseOpts<F>,
+    data: option::IntoIter<String>,
+}
 
+impl Html  {
+    pub fn from_string(string: &str) -> Html {
+        Html {
+            opts: ParseOpts::default(),
+            data: Some(string.to_string()).into_iter(),
+        }
+    }
 
-pub fn parse<'a, F, I>(source: I, arena: &'a Arena<Node<'a>>, opts: ParseOpts<F>) -> &'a Node<'a>
-                       where I: IntoIterator<Item=String>, F: FnMut(Cow<'static, str>) {
-    let parser = Parser {
-        arena: arena,
-        document_node: Node::new_document(arena),
-        on_parse_error: opts.on_parse_error,
-    };
-    let opts = html5ever::ParseOpts {
-        tokenizer: opts.tokenizer,
-        tree_builder: opts.tree_builder,
-    };
-    let parser = html5ever::parse_to(parser, source.into_iter(), opts);
-    parser.document_node
+    pub fn from_file<P: AsRef<Path>>(path: P) -> Html {
+        let mut buf = String::new();
+        let mut file = File::open(&path).unwrap();
+        file.read_to_string(&mut buf).unwrap();
```

Comment:
This should return a `Result` and use `try!` instead of `unwrap()`.

Ideally it should also read the file incrementally (by chunks of a few KB each) but the html5ever currently does not allow easily propagating I/O errors, so `read_to_string` is fine for now.